db/migrate/20171122121841_change_event_log_user_agent_id_to_integer.rb

```diff
@@ -1,9 +1,21 @@
 class ChangeEventLogUserAgentIdToInteger < ActiveRecord::Migration[5.1]
   def up
-    change_column :event_logs, :user_agent_id, "BIGINT"
+    Lhm.cleanup(:run)
+    Lhm.change_table :event_logs do |m|
+      m.remove_foreign_key :event_logs, :user_agents
+      m.change_column :user_agents, :id, "BIGINT"
+      m.change_column :event_logs, :user_agent_id, "BIGINT"
+      m.add_foreign_key :event_logs, :user_agents
+    end
```

Comment:
How come we need to do this? and not change any of the other id types / foreign keys?

Reply:
When originally adding that foreign key, it was added by adding a column then specifying the key, so the type was explicitly set to `INT` ( https://github.com/alphagov/signon/blob/master/db/migrate/20170908160100_add_ip_address_and_user_agent_id_to_event_logs.rb ) - this needs to change now due to the default value of IDs changing to BIGINT - so db:schema:load fails due to the mismatched data type. 

This is the only place this error occurs

Reply:
Ah I see, yes - as that is the only foreign key.

Ok so I don't think we should be doing that via the method you have there. As that ends us in up a situation where running rake `db:setup` has a different database schema to what is on production and you'll hit same problems with any future foreign keys.

Rails actually has a system for this where it specifies id type as part of the create table argument: `create_table "batch_invitation_users", id: :integer`

If I run db:schema:dump with rails 5 on this I get this https://gist.github.com/kevindew/4de510ce86eff45622d1aa1d2d11ce09 which will load up without any need for migrations. Does however create a "lhma_2017_09_18_16_13_31_585_event_logs" table that may not be desirable.

Reply:
It seems the `create_table, id: :integer` method is the simplest way to go :+1: